File: user/src/main/scala/ru/tdd/app/models/dto/BooleanDTO.scala
package ru.tdd.app.models.dto

import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}

/**
 * @author Tribushko Danil
 * @since 20.12.2025
 */
case class BooleanDTO(result: Boolean)

object BooleanDTO {

  implicit val encoder: JsonEncoder[BooleanDTO] = DeriveJsonEncoder.gen[BooleanDTO]
  implicit val decoder: JsonDecoder[BooleanDTO] = DeriveJsonDecoder.gen[BooleanDTO]
}

File: telegram-bot/src/main/java/ru/tdd/telegram_bot/model/dto/BooleanDTO.java
package ru.tdd.telegram_bot.model.dto;

/**
 * @author Tribushko Danil
 * @since 20.12.2025
 */
public class BooleanDTO {

    private boolean result;

    public BooleanDTO(){}

    public BooleanDTO(Boolean result) {
        this.result = result;
    }

    public boolean getResult() {
        return result;
    }

    public void setResult(boolean result) {
        this.result = result;
    }
}
